### PR TITLE
MEN-3174 2FA Verify token backend tests

### PR DESCRIPTION
**Pull request by merlin-northern**
ChangeLog:title
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

File: testutils/api/useradm.py [hunk 1]
```diff
@@ -19,5 +19,6 @@
 URL_LOGIN = '/auth/login'
 URL_SETTINGS = '/settings'
 URL_2FAQR = '/2faqr'
+URL_VERIFY = '/2faverify'
```
Comment (lluiscampos) [inline testutils/api/useradm.py]: To follow current convention, I think this one should be `URL_2FAVERIFY`

File: backend-tests/tests/test_useradm_enterprise.py [hunk 2]
```diff
@@ -71,6 +71,12 @@ def _login(self, user, totp=None):
         r = uadm.call("POST", useradm.URL_LOGIN, auth=(user.name, user.pwd), body=body)
         return r
 
+    def _verify(self, utoken, totp):
```
Comment (lluiscampos) [inline backend-tests/tests/test_useradm_enterprise.py]: This method is only used once, consider removing it

File: backend-tests/tests/test_useradm_enterprise.py [hunk 3]
```diff
@@ -124,10 +122,25 @@ def test_enable_disable(self, tenants_users):
...
 
+        # verify token
+        r = self._verify(user_2fa_tok, tok)
+        assert r.status_code == 202
+
         # login with totp succeeds
         r = self._login(user_2fa, totp=tok)
         assert r.status_code == 200
 
+        r = self._login(user_2fa)
```
Comment (lluiscampos) [inline backend-tests/tests/test_useradm_enterprise.py]: I would add here the comment partially removed above: "token has been verified, login shall now work anymore" (or something like that)

**Comment by lluiscampos** [comment]
See also https://github.com/mendersoftware/useradm-enterprise/pull/27

**Comment by merlin-northern** [comment]
> See also [mendersoftware/useradm-enterprise#27](https://github.com/mendersoftware/useradm-enterprise/pull/27)

it goes with https://github.com/mendersoftware/useradm-enterprise/pull/30

File: backend-tests/tests/test_useradm_enterprise.py (hunk 3 above)
Comment (merlin-northern) [reply backend-tests/tests/test_useradm_enterprise.py]: added.

File: backend-tests/tests/test_useradm_enterprise.py (hunk 2 above)
Comment (merlin-northern) [reply backend-tests/tests/test_useradm_enterprise.py]: I am sorry, but please lets keep it, it is logical and cleaner this way.

File: testutils/api/useradm.py (hunk 1 above)
Comment (merlin-northern) [reply testutils/api/useradm.py]: changed.

**Comment by merlin-northern** [comment]
> Changelogs are for user-facing changes. Please use `changelog: none` for commits that only modify tests.
> 
> I'll let someone else review the actual code.

sure. changed to none.

File: backend-tests/tests/test_useradm_enterprise.py (hunk 2 above)
Comment (lluiscampos) [reply backend-tests/tests/test_useradm_enterprise.py]: :+1:

File: backend-tests/tests/test_useradm_enterprise.py (hunk 3 above)
Comment (lluiscampos) [reply backend-tests/tests/test_useradm_enterprise.py]: s/logi/login :)

File: backend-tests/tests/test_useradm_enterprise.py [hunk 4]
```diff
@@ -102,19 +108,11 @@ def test_enable_disable(self, tenants_users):
...
+        # enable tfa for 1 user, straight login still works, token is not verified
         self._toggle_tfa(user_2fa_tok, on=True)
         r = self._login(user_2fa)
-        assert r.status_code == 401
-
-        # the other user, and other tenant's users, are unaffected
-        r = self._login(user_no_2fa)
         assert r.status_code == 200
 
-        for other_user in tenants_users[1].users:
```
Comment (mchalski) [inline backend-tests/tests/test_useradm_enterprise.py]: you can leave this check in

**Pull request merged**